baidupcsapi/api.py

```diff
@@ -690,6 +690,166 @@ def get_url(dlink):
+                    "delete_fs_id": "0",
+                    "extent_int3": "0",
+                    "extent_tinyint1": "0",
+                    "extent_tinyint2": "0",
+                    "extent_tinyint3": "0",
+                    "extent_tinyint4": "0",
+                    "path": "\/\u65b0\u5efa\u6587\u4ef6\u5939(1)\/[SumiSora][Yosuga_no_Sora][BDRip][BIG5][720P]",
+                    "root_ns": 465254146,
+                    "md5": "",
+                    "file_key": ""
+                }]
+            },
+            "loginstate": 0,
+            "channel": 4,
+            "third_url": 0,
+            "bdstoken": null ,
+            "sampling": {
+                "expvar": ["chengyong"]
+            },
+            "is_vip": 0,
+            "description": "",
+            "shorturl": "1skhBegP",
+            "shareinfo": "",
+            "is_baiduspider": 0,
+            "isinwhitelist": 0,
+            "public": 0,
+            "shareid": 23915657,
+            "bj_unicom": 0,
+            "visitor_uk": null ,
+            "visitor_avatar": null ,
+            "timestamp": 1458198232,
+            "sign": "40398487e416230080a32ea607ac686f88bfab25",
+            "sekey": "nvIaITp2vb68JXUwMyQp4DmtFeLO8642",
+            "novelid": false,
+            "is_master_vip": 0,
+            "urlparam": [],
+            "XDUSS": "null"
+        }
+        """
+        respond = self._request(None, url=url)
+
+        target_url = respond.url
+        shareid, uk = None, None
+        m = re.search(r"shareid=(\d+)", target_url)
+        if m:
+            shareid = m.group(1)
+        m = re.search(r"uk=(\d+)", target_url)
+        if m:
+            uk = m.group(1)
+
+        # 检查验证码, 如果成功, 当前用户就被授权直接访问资源了
+        if password:
+            verify_result = self._verify_shared_file(shareid, uk, password)
+            if not verify_result or verify_result['errno'] != 0:
+                print "PCS.get_shared_file_list(), verify code error!"
+                exit(-1)
+
+        # 从html中解析文件列表
+        html = self._request(None, url=target_url).content
+        r = re.compile(r".*_context =(.*);.*")
+        m = r.search(html)
+        if m:
+            context = json.loads(m.group(1))
+            file_list = context['file_list']['list']
+            uk = str(context['uk'])
+            shareid = str(context['shareid'])
+            ret = {"filelist": []}
+            for f in file_list:
+                file_obj = {
+                    'filename': f['server_filename'],
+                    'size': f['size'],
+                    'isdir': f['isdir']
+                }
+                if not filter or filter(file_obj):
+                    ret['filelist'].append(f['path'])
+            ret['result'] = self._save_shared_file_list(shareid, uk, path, ret['filelist'])
+            return ret
+        else:
+            # 获取文件列表失败
+            print "get_share_filelist failed"
+            exit(-1)
```

Comment:
好的，我改一下

On Mar 19, 2016 11:53 PM, "latyas(懒)" <notifications@github.com> wrote:

In baidupcsapi/api.py<https://github.com/ly0/baidupcsapi/pull/27#discussion_r56750683>:

> +            shareid = str(context['shareid'])
> +            ret = {"filelist": []}
> +            for f in file_list:
> +                file_obj = {
> +                    'filename': f['server_filename'],
> +                    'size': f['size'],
> +                    'isdir': f['isdir']
> +                }
> +                if not filter or filter(file_obj):
> +                    ret['filelist'].append(f['path'])
> +            ret['result'] = self._save_shared_file_list(shareid, uk, path, ret['filelist'])
> +            return ret
> +        else:
> +            # 获取文件列表失败
> +            print "get_share_filelist failed"
> +            exit(-1)


建议抛自定义异常而不要直接exit呐

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/ly0/baidupcsapi/pull/27/files/8851b46885d2d431e589f1c87379d8ab5628cfd7#r56750683>